src/js/game/buildings/analyzer.js

```diff
@@ -68,12 +137,74 @@ export class MetaAnalyzerBuilding extends MetaBuilding {
                     },
                 ],
             })
-        );
+        ),
 
+    entity =>
         entity.addComponent(
             new LogicGateComponent({
                 type: enumLogicGateType.analyzer,
             })
-        );
-    }
-}
+        ),
+];
+
+MetaAnalyzerBuilding.silhouetteColors = {

```

Comment:
I don't like the mapping for variants here. Why not make a variant its own class?

Reply:
Have done that in https://github.com/DJ1TJOO/shapez.io/tree/modloader-variant-classes. Note this is missing a lot. It's just for feedback right now

Reply:
Also fixed circular dependencie for **defaultVariantBuilding**

Reply:
which files should I check there?

Reply:
https://github.com/DJ1TJOO/shapez.io/blob/modloader-variant-classes/src/js/game/meta_building.js
https://github.com/DJ1TJOO/shapez.io/blob/modloader-variant-classes/src/js/game/meta_building_variant.js
https://github.com/DJ1TJOO/shapez.io/blob/modloader-variant-classes/src/js/game/meta_building_registry.js

Examples:
https://github.com/DJ1TJOO/shapez.io/tree/modloader-variant-classes/src/js/game/buildings/variants
https://github.com/DJ1TJOO/shapez.io/blob/modloader-variant-classes/src/js/game/buildings/analyzer.js
https://github.com/DJ1TJOO/shapez.io/blob/modloader-variant-classes/src/js/game/buildings/balancer.js

Reply:
The change itself looks nice, although very breaking. Let's start with the modloader only